### PR TITLE
Add rust locals.scm

### PR DESCRIPTION
**Pull request by theHamsta**
- [x] definitions (started)
- [x] scopes

I will try to use the existing work in the discussion #57

### CODE REVIEW

**Comment by theHamsta** [comment]
Probably still incomplete, will play around with it another day.

File: queries/rust/locals.scm [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+(field_declaration
+  name: (field_identifier) @definition.field)
+
+(enum_variant
+  name: (identifier) @definition.field) 
+
+; References
+(identifier) @reference
+(type_identifier) @reference
+(field_identifier) @reference
```
Comment (theHamsta) [inline queries/rust/locals.scm:74]: Same issue as in #164. I think we should also consider references to types and fields.

Comment (vigoux) [reply queries/rust/locals.scm:74]: I agree on this in the end

File: queries/cpp/locals.scm [hunk 2]
```diff
@@ -20,6 +20,13 @@
   name: (scoped_type_identifier
           name: (type_identifier) @definition.type))
 
+;; Namespaces
+(namespace_definition 
+  name: (identifier) @definition.namespace
+  body (_) @scope)
+
+(namespace_identifier) @reference
+
```
Comment (vigoux) [inline queries/cpp/locals.scm:29]: Same for this

File: queries/cpp/locals.scm [hunk 3]
```diff
@@ -49,3 +56,4 @@
 
 (destructor_name
   name: (_) @constructor)
+
```
Comment (vigoux) [inline queries/cpp/locals.scm:59]: Same

File: queries/rust/locals.scm [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+(for_expression) @scope
+(loop_expression) @scope
+(if_expression) @scope
+(if_let_expression) @scope
+(match_expression) @scope
+(match_arm) @scope
+
+(struct_item) @scope
+(enum_item) @scope
+(impl_item) @scope
```
Comment (vigoux) [inline queries/rust/locals.scm:100]: Can we refactor that in  a `[ ]` block ?

File: queries/rust/locals.scm [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+(field_identifier) @reference
+
+
+; Macros
+(macro_definition
+  name: (identifier) @definition.macro)
+
+; Module
+(mod_item 
+  name: (identifier) @definition.namespace)
```
Comment (vigoux) [inline queries/rust/locals.scm:83]: Now I see why you added those namespaces :wink:

File: queries/rust/locals.scm (hunk 1 above)
Comment (theHamsta) [reply queries/rust/locals.scm:74]: Should it be `@reference.type`?

Comment (vigoux) [reply queries/rust/locals.scm:74]: Yes, I like that more !

**Comment by vigoux** [comment]
Awesome ! That is one step further in #52  !

**Pull request merged**